bridges/snowbridge/pallets/inbound-queue-v2/src/lib.rs

```diff
@@ -233,13 +233,17 @@ pub mod pallet {
 			let (nonce, relayer_fee) = (message.nonce, message.relayer_fee);
 
 			// Verify the message has not been processed
-			ensure!(!Nonce::<T>::get(nonce.into()), Error::<T>::InvalidNonce);
+			ensure!(!Nonce::<T>::get(nonce), Error::<T>::InvalidNonce);
 
 			let xcm =
 				T::MessageConverter::convert(message).map_err(|error| Error::<T>::from(error))?;
 
 			// Forward XCM to AH
 			let dest = Location::new(1, [Parachain(T::AssetHubParaId::get())]);
+
+			// Mark message as received
+			Nonce::<T>::set(nonce);
```

Comment:
You mentioned you want messages to be able to be retried so there's no loss of user funds. Does marking the message as received before `send_xcm` allow it to be retried in the future if `send_xcm` fails?